src/Servers/Kestrel/Core/src/Internal/Http3/Http3Stream.cs

```diff
@@ -409,6 +409,8 @@ private bool TryClose()
                 try
                 {
                     await _frameWriter.CompleteAsync();
+
+                    await _context.StreamContext.DisposeAsync();
```

Comment:
Can Http3Writer.CompleteAsync() really throw? If so, this should probably be in a finally. If not, we should probably remove this try/catch/finally altogether. DisposeAsync() shouldn't throw.